ztp/resource-generator/src/watcher.py

```diff
@@ -164,6 +165,41 @@ def __init__(self, api_response, resourcename="siteconfigs", debug=False):
                     shutil.rmtree(self.tmpdir)
                     shutil.rmtree(out_tmpdir)
 
+    # Note: this solution is limited to SNO (one cluster per siteconfig).
+    def _handle_site_deletions(self) -> bool:
+        del_siteconfig_list = os.listdir(self.del_path)
+        for item in del_siteconfig_list:
+            with open(os.path.join(self.del_path, item), "r") as yi:
+                obj = yaml.safe_load(yi)
+            if obj.get("kind", {}) == "SiteConfig":
+                # Assuming here that name and namespace of the
+                # managedcluster are identical
+                name = obj.get("metadata", {}).get("namespace")
+                try:
+                    status = None
+                    cmd = ["oc", "delete", f"managedcluster/{name}",
+                           "-n", f"{name}"]
```

Comment:
the managedCluster is a cluster scope resource no namespace for it

Reply:
Fixed